app/src/main/java/org/schabi/newpipe/fragments/list/search/filter/SearchFilterDialogSpinnerAdapter.java

```diff
@@ -0,0 +1,220 @@
+// Created by evermind-zz 2022, licensed GNU GPL version 3 or later
+
+package org.schabi.newpipe.fragments.list.search.filter;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.view.Gravity;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.Spinner;
+import android.widget.TextView;
+
+import org.schabi.newpipe.extractor.search.filter.FilterContainer;
+import org.schabi.newpipe.extractor.search.filter.FilterGroup;
+import org.schabi.newpipe.extractor.search.filter.FilterItem;
+import org.schabi.newpipe.util.ServiceHelper;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.schabi.newpipe.fragments.list.search.filter.SearchFilterOptionMenuAlikeDialogGenerator.dipToPixels;
+
+public class SearchFilterDialogSpinnerAdapter extends BaseAdapter {
+
+    private final Context context;
+    private final FilterGroup group;
+    private final BaseSearchFilterUiGenerator.UiWrapperMapDelegate wrapperDelegate;
+    private final Spinner spinner;
+    private final Map<Integer, Integer> id2PosMap = new HashMap<>();
+    private final Map<Integer, UiItemWrapperSpinner>
+            viewWrapperMap = new HashMap<>();
+
+    public SearchFilterDialogSpinnerAdapter(
+            final Context context,
+            final FilterGroup group,
+            final BaseSearchFilterUiGenerator.UiWrapperMapDelegate wrapperDelegate,
+            final Spinner filterDataSpinner) {
+        this.context = context;
+        this.group = group;
+        this.wrapperDelegate = wrapperDelegate;
+        this.spinner = filterDataSpinner;
+
+        createViewWrappers();
+    }
+
+    @Override
+    public int getCount() {
+        return group.getFilterItems().length;
+    }
+
+    @Override
+    public Object getItem(final int position) {
+        return group.getFilterItems()[position];
+    }
+
+    @Override
+    public long getItemId(final int position) {
+        return position;
+    }
+
+    @Override
+    public View getView(final int position, final View convertView, final ViewGroup parent) {
+        final FilterItem item = group.getFilterItems()[position];
+        final TextView view;
+
+        if (convertView != null) {
+            view = (TextView) convertView;
+        } else {
+            view = createViewItem();
+        }
+
+        initViewWithData(position, item, view);
+        return view;
+    }
+
+    @SuppressLint("WrongConstant")
+    private void initViewWithData(final int position,
+                                  final FilterItem item,
+                                  final TextView view) {
+        final UiItemWrapperSpinner wrappedView =
+                viewWrapperMap.get(position);
+        Objects.nonNull(wrappedView);
```

Comment:
I'm assuming this is a non-null enforcement check.

```suggestion
        Objects.requireNonNull(wrappedView);
```